File: app/src/main/java/net/emilla/math/FloatingPointNumber.java
package net.emilla.math;

final class FloatingPointNumber implements CalcValue<Double>, ArithmeticToken {
    private final double mValue;

    FloatingPointNumber(double value) {
        mValue = value;
    }

    @Override
    public Double get() {
        return mValue;
    }
}

File: app/src/main/java/net/emilla/math/FloatingPointNumber.kt
package net.emilla.math

import androidx.annotation.StringRes
import net.emilla.math.CalcToken.InfixToken

internal class FloatingPointNumber(num: String, @StringRes errorTitle: Int) : InfixToken {
    @JvmField
    val value: Double = tryParseDouble(num, errorTitle)
}
